### PR TITLE
Cmd socat

### PR DESCRIPTION
**Pull request by AdriannaPienkowska**

### CODE REVIEW

**Comment by Ernold11** [comment]
Check List
----------
* [ ]  CI is passed
* [ ]  master is merged into current branch
* [ ]  public methods should be at the top of class, protected at bottom  
       example: `on_new_line()` is public, `_parse()` is protected
* [ ]  Name of methods should tell what its doing, not how are doing  
       example: `_parse_re_via_dev_proto_metric()` -> `_parse_via_dev_proto_metric()`
* [ ]  All parsing lines (on_new_line) should be covered by unit tests
* [ ]  Be sure that  COMMAND_OUTPUT_* / COMMAND_KWARGS_* / COMMAND_RESULT_* cover all parsing lines
* [ ]  Parsed line example, parsing regexp, parsing method should be next to each other
* [ ]  Using named groups inside regexp (like `r"^\s*(?P<ADDRESS>\S+)\s+via\s+(?P<VIA>\S+)”`) is preferred
* [ ]  Longer regexp are before shorter regexp
* [ ]  Resources for pytest should be declared as `@pytest.fixture` or `@pytest.yield_fixture`
* [ ]  There are no sensitive data (company specific) like: IP addr, passwords, commands, tools etc

File: moler/cmd/unix/socat.py [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+__email__ = 'adrianna.pienkowska@nokia.com'
+
+from moler.cmd.unix.genericunix import GenericUnixCommand
+from moler.exceptions import CommandFailure
+from moler.exceptions import ParsingDone
+import re
+
+
+class Socat(GenericUnixCommand):
+    def __init__(self, connection, input_options=None, output_options=None, options=None):
```
Comment (Ernold11) [inline moler/cmd/unix/socat.py]: Add prompt and newline chars

File: moler/cmd/unix/socat.py [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+
+COMMAND_RESULT_both_types = {
+    'RESULT': ['ssh: Could not resolve hostname server1.nixcraft.net.in: Name or service not known'],
+    'INFO': ['W open("/dev/tty", O_NOCTTY, 0640): No such device or address',
+             'W waitpid(): child 31539 exited with status 255']
+}
+
+
+COMMAND_OUTPUT_version = """
+xyz@debian:~$ socat -V
```
Comment (Ernold11) [inline moler/cmd/unix/socat.py]: Please add parsing for that output, i.e. dict with `WITH_STDIO ` as key and `1` as value

**Comment by AdriannaPienkowska** [comment]
I have just finished changes.

File: moler/cmd/unix/socat.py [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+from moler.cmd.unix.genericunix import GenericUnixCommand
+from moler.exceptions import CommandFailure
+from moler.exceptions import ParsingDone
+import re
+
+
+class Socat(GenericUnixCommand):
+    def __init__(self, connection, input_options=None, output_options=None, options=None, prompt=None,
+                 new_line_chars=None):
+        super(Socat, self).__init__(connection=connection, prompt=None, new_line_chars=None)
```
Comment (Ernold11) [inline moler/cmd/unix/socat.py]: Please use values passed into command constructor!

**Pull request merged**